### PR TITLE
AMQNET-617: Support Message Selectors

### PR DESCRIPTION
**Pull request by HavretGC**
This PR adds support for message selectors.

### CODE REVIEW

**Comment by Havret** [comment]
@cjwmorgan-sol Could you please take a look at this PR?

**Comment by cjwmorgan-sol** [comment]
Looks good to me. I verified the tests against a broker and they passed.

File: README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ So in general most of the top level classes that implement the Apache.NMS interf
 | IMessageProducer | Y * | Anonymous producers are only supported on connections with the ANONYMOUS-RELAY capability. |
 | MsgDeliveryMode.Persistent | Y | Producers will block on send until an outcome is received or will timeout after waiting the RequestTimeout timespan amount. Exceptions may be throw depending on the outcome or if the producer times out. |
 | MsgDeliveryMode.NonPersistent | Y | Producers will not block on send nor expect to receive an outcome. Should an exception be raised from the outcome the exception will be delivered using the the connection ExceptionListener. |
-| IMessageConsumer | Y * | Message Selectors and noLocal filter are not supported. |
+| IMessageConsumer | Y * | NoLocal filter is not supported. |
```
Comment (michaelandrepearce) [inline README.md]: can we add a like showing that message selectors are supported

Comment (michaelpearce-gain) [reply README.md]: just re-read, in context its right.

**Pull request merged**